src/vs/workbench/contrib/extensions/browser/extensionsWorkbenchService.ts

```diff
@@ -1043,6 +1056,36 @@ export class ExtensionsWorkbenchService extends Disposable implements IExtension
 		}));
 	}
 
+	private isAutoUpdateEnabled(): boolean {
+		return this.getAutoUpdateValue() !== false;
+	}
+
+	getAutoUpdateValue(): AutoUpdateConfigurationValue {
+		const autoUpdate = this.configurationService.getValue<AutoUpdateConfigurationValue>(AutoUpdateConfigurationKey);
+		if (<any>autoUpdate === 'onlySelectedExtensions') {
+			return false;
+		}
+		return isBoolean(autoUpdate) || autoUpdate === 'onlyEnabledExtensions' ? autoUpdate : true;
+	}
+
+	async updateAutoUpdateValue(value: AutoUpdateConfigurationValue): Promise<void> {
+		const wasEnabled = this.isAutoUpdateEnabled();
+		const isEnabled = value !== false;
+		if (wasEnabled !== isEnabled) {
+			const result = await this.dialogService.confirm({
+				title: nls.localize('confirmEnableDisableAutoUpdate', "Auto Update Extensions"),
+				message: isEnabled
+					? nls.localize('confirmEnableAutoUpdate', "Do you want to enable auto update for all extensions?")
+					: nls.localize('confirmDisableAutoUpdate', "Do you want to disable auto update for all extensions?"),
+				detail: nls.localize('confirmEnableDisableAutoUpdateDetail', "This will reset the auto update settings you have set for individual extensions."),
```

Comment:
Wording nit: It think it would sound better if you change `the` to `any` as it indicates that you may or may not have set the setting for individual extensions

```suggestion
				detail: nls.localize('confirmEnableDisableAutoUpdateDetail', "This will reset any auto update settings you have set for individual extensions."),
```